### PR TITLE
Single Server

### PR DESCRIPTION
**Pull request by benbjohnson**
- [x] CHANGELOG.md updated
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: client/v2/client.go [hunk 1]
```diff
@@ -330,7 +330,7 @@ func (bp *batchpoints) SetRetentionPolicy(rp string) {
 
 // Point represents a single data point
 type Point struct {
-	pt models.Point
```
Comment (jwilder) [inline client/v2/client.go]: Why this change?  `pt` was intentionally not exported for v2 client I believe.

File: cmd/influxd/run/server.go [hunk 2]
```diff
@@ -137,16 +127,17 @@ func NewServer(c *Config, buildInfo *BuildInfo) (*Server, error) {
...
 	}
 
+	// In 0.11 we removed MetaServers from node.json.  To avoid confusion for
+	// existing users, force a re-save of the node.json file to remove that property
+	// if it happens to exist.
+	nodeContents, err := ioutil.ReadFile(filepath.Join(c.Meta.Dir, "node.json"))
+	if err == nil && strings.Contains(string(nodeContents), "MetaServers") {
+		node.Save()
+	}
+
```
Comment (jwilder) [inline cmd/influxd/run/server.go]: This code was removed in #5957 because it breaks rolling back to `0.10` if necessary.

File: cmd/influxd/run/server_test.go [hunk 3]
```diff
@@ -494,43 +495,10 @@ func TestServer_Write_LineProtocol_Integer(t *testing.T) {
 	}
 }
 
-// Ensure the server returns a partial write response when some points fail to parse. Also validate that
-// the successfully parsed points can be queried.
-func TestServer_Write_LineProtocol_Partial(t *testing.T) {
```
Comment (jwilder) [inline cmd/influxd/run/server_test.go]: This test shouldn't be removed.  It is testing that multiple points in a line protocol batch doesn't fail the whole batch if one fails to parse.

File: services/httpd/handler.go [hunk 4]
```diff
@@ -117,14 +118,6 @@ func NewHandler(requireAuthentication, loggingEnabled, writeTrace, JSONWriteEnab
 			"ping-head",
 			"HEAD", "/ping", true, true, h.servePing,
 		},
-		route{ // Ping w/ status
```
Comment (jwilder) [inline services/httpd/handler.go]: Hmm.  I think we should keep these for now and log a warning that they are deprecated.   We should log a message that they should use `/ping` instead.  Any load balancer configured to point to these is going to think the host is down and i won't be easy to determine why.

File: services/meta/client.go [hunk 5]
```diff
@@ -957,6 +833,15 @@ func (c *Client) WaitForDataChanged() chan struct{} {
 	return c.changed
 }
 
+// commit assumes it is under a full lock
+func (c *Client) commit(data *Data) {
+	data.Index++
+	c.cacheData = data
+	c.Snapshot()
```
Comment (jwilder) [inline services/meta/client.go]: error is ignored.  Should be returned if we can't save.  Also, should probably try to snapshot first and then update in-memory state.

File: client/v2/client.go (hunk 1 above)
Comment (e-dard) [inline client/v2/client.go]: This was me. I needed this so I could easily create `client.Point` from an existing `models.Point`. The current `NewPoint` function relies on all the separate aspects of a point, which seemed clunky for the way I wanted to use the client.

File: cmd/influxd/run/server.go (hunk 2 above)
Comment (corylanou) [inline cmd/influxd/run/server.go]: @jwilder are you saying it should be removed now, or left in?

File: client/v2/client.go (hunk 1 above)
Comment (jwilder) [inline client/v2/client.go]: The problem is that the v2 client was intended to have points be immutable.  Exporting the underlying Point defeats the purpose of that.

What about adding a function like:
```
NewPointFrom(pt models.Point) {
  return &Point{pt:pt}
}
```

Comment (e-dard) [inline client/v2/client.go]: OK will do!

**Comment by jwilder** [comment]
:+1:

**Comment by dgnorton** [comment]
+1

**Comment by benbjohnson** [comment]
:+1:

**Comment by toddboom** [comment]
:boom:
